packages/typescript-estree/src/ts-estree/estree-to-ts-node-types.ts

```diff
@@ -77,7 +77,7 @@ export interface EstreeToTsNodeTypes {
     | ts.Token<ts.SyntaxKind.ImportKeyword | ts.SyntaxKind.NewKeyword>;
   [AST_NODE_TYPES.PrivateIdentifier]: ts.PrivateIdentifier;
   [AST_NODE_TYPES.IfStatement]: ts.IfStatement;
-  [AST_NODE_TYPES.ImportAttribute]: ts.AssertEntry;
+  [AST_NODE_TYPES.ImportAttribute]: ts.ImportAttribute;
```

Comment:
Hmm - should we keep both for now...? Can we do one of those "check for TS version range" conditional type magics to pick the right type?

Reply:
I tried a fancy `keyof`: https://github.com/typescript-eslint/typescript-eslint/pull/7821/commits/ac4beddeda21a1a15e228471e1e293c270b25cff#diff-6ce24a606b5c67a1a4bb3dbf7060e040e84bd0de5250ec312d8a2ddad931eef0. Is that what you're looking for?

I don't remember previous cases of trying this.